src/objset.c

```diff
@@ -844,9 +835,13 @@ static Obj FuncCLEAR_OBJ_SET(Obj self, Obj set) {
  *  GAP function to return values in set as a list.
  */
 
-static Obj FuncOBJ_SET_VALUES(Obj self, Obj set) {
-  CheckArgument("OBJ_SET_VALUES", set, T_OBJSET, -1);
-  return ObjSetValues(set);
+static Obj FuncOBJ_SET_VALUES(Obj self, Obj set)
+{
+    RequireArgumentCondition("OBJ_SET_VALUES", set, "objset",
+                             TNUM_OBJ(set) == T_OBJSET,
+                             "must be a mutable object set");
```

Comment:
@rbehrends is there any reason this function requires mutable objset?

Reply:
No, this and the other checks are probably the result of copy-and-paste errors.